rspace/src/main/scala/coop/rchain/rspace/history/RadixTree.scala

```diff
@@ -193,15 +206,23 @@ object RadixTree {
 
           // Decoding type of non-empty item
           val item =
-            if (isLeaf(secondByte)) Leaf(ByteVector(prefix), ByteVector(valOrPtr))
-            else NodePtr(ByteVector(prefix), ByteVector(valOrPtr))
+            if (isLeaf(secondByte))
+              Leaf(KeyPath.create(ByteVector(prefix)), Blake2b256Hash.fromByteArray(valOrPtr))
+            else
+              NodePtr(KeyPath.create(ByteVector(prefix)), Blake2b256Hash.fromByteArray(valOrPtr))
```

Comment:
If KeyPath is abstraction over internal binary structure, the whole RadixTree should be dependent only on KeyPath without using ByteVector or any other binary structure directly.